src/plexosdb/sqlite.py

```diff
@@ -459,14 +461,17 @@ def add_object(
 
         params = (object_name, class_id, category_id, str(uuid.uuid4()), description)
         placeholders = ", ".join("?" * len(params))
-        with self._conn as conn:
-            cursor = conn.cursor()
-            cursor.execute(
-                f"INSERT INTO {Schema.Objects.name}(name, class_id, category_id, GUID, description) "
-                f"VALUES({placeholders})",
-                params,
-            )
-            object_id = cursor.lastrowid
+        try:
+            with self._conn as conn:
+                cursor = conn.cursor()
+                cursor.execute(
+                    f"INSERT INTO {Schema.Objects.name}(name, class_id, category_id, GUID, description) "
+                    f"VALUES({placeholders})",
+                    params,
+                )
+                object_id = cursor.lastrowid
+        except sqlite3.IntegrityError as err:
```

Comment:
I think it might be worth that we manually show what error it is.

Reply:
I was trying to maintain the ValueError for consistency with previous get_object behavior. I also like the idea that a module using plexosdb could catch an error from PlexosDB.add_object without importing sqlite3. I am happy to remove this error conversion though.